### ISSUE TITLE
Filename is ignored in middleware mode; default filename is always used 

### ISSUE THREAD

**Issue by SherModo** [tmm1/stackprof#94]
https://github.com/tmm1/stackprof/blob/master/lib/stackprof/middleware.rb#L27
https://github.com/tmm1/stackprof/blob/master/lib/stackprof/middleware.rb#L43
https://github.com/tmm1/stackprof/blob/master/lib/stackprof/middleware.rb#L46

**Comment by SherModo** [comment]
I made these changes to make it work locally:
```
def initialize(app, options = {})
      @app       = app
      @options   = options
      @num_reqs  = options[:save_every] || nil

      Middleware.mode     = options[:mode] || :cpu
      Middleware.interval = options[:interval] || 1000
      Middleware.raw      = options[:raw] || false
      Middleware.enabled  = options[:enabled]
      Middleware.out      = options[:out] || nil
      Middleware.path     = 'tmp'
      at_exit{ Middleware.save } if options[:save_at_exit]
    end
```

```
def save
        if results = StackProf.results
          if Middleware.out
            filename = File.basename(Middleware.out)
            path = File.dirname(Middleware.out)
          else
            filename = "stackprof-#{results[:mode]}-#{Process.pid}-#{Time.now.to_i}.dump"
            path = Middleware.path
          end
          FileUtils.mkdir_p(path)
          File.open(File.join(Middleware.path, filename), 'wb') do |f|
            f.write Marshal.dump(results)
          end
          filename
        end
      end
```

**Comment by tmm1** [comment]
Please submit a pull request.

**Comment by itsderek23** [comment]
Fixed by #95.
